cub/test/catch2_test_device_histogram.cu

```diff
@@ -0,0 +1,647 @@
+{
+  // TODO(bgruber): eventually, just pick a random number of levels per channel
+
+  // first channel tests maximum number of levels, later channels less and less
+  array<int, ActiveChannels> r{max_level_count};
+  for (size_t c = 1; c < ActiveChannels; ++c)
+  {
+    r[c] = r[c - 1] / 2 + 1;
+  }
+  return r;
+}
+
+struct bit_and_anything
+{
+  template <typename T>
+  _CCCL_HOST_DEVICE auto operator()(const T& a, const T& b) const -> T
+  {
+    using U = typename cub::Traits<T>::UnsignedBits;
+    return c2h::bit_cast<T>(static_cast<U>(c2h::bit_cast<U>(a) & c2h::bit_cast<U>(b)));
+  }
+};
+
+template <typename SampleT, int Channels, size_t ActiveChannels, typename CounterT, typename LevelT, typename OffsetT>
+void test_even_and_range(LevelT max_level, int max_level_count, OffsetT width, OffsetT height, int entropy_reduction = 0)
+{
+  const auto padding_bytes = static_cast<OffsetT>(GENERATE(size_t{0}, 13 * sizeof(SampleT)));
+  CAPTURE(
+    c2h::type_name<SampleT>(),
+    c2h::type_name<CounterT>(),
+    c2h::type_name<LevelT>(),
+    c2h::type_name<OffsetT>(),
+    Channels,
+    ActiveChannels,
+    CoutCast(max_level),
+    max_level_count,
+    width,
+    height,
+    padding_bytes,
+    entropy_reduction);
+
+  // Prepare input image (samples)
+  const OffsetT row_pitch = width * Channels * sizeof(SampleT) + padding_bytes;
+  const auto num_levels   = generate_level_counts_to_test<ActiveChannels>(max_level_count);
+
+  const OffsetT total_samples = height * (row_pitch / sizeof(SampleT));
+  c2h::device_vector<SampleT> d_samples;
+  d_samples.resize(total_samples);
+
+  if (entropy_reduction >= 0)
+  {
+    c2h::gen(CUB_SEED(1), d_samples, SampleT{0}, static_cast<SampleT>(max_level));
+    if (entropy_reduction > 0)
+    {
+      c2h::device_vector<SampleT> tmp(d_samples.size());
+      for (int i = 0; i < entropy_reduction; ++i)
+      {
+        c2h::gen(CUB_SEED(1), tmp);
+        thrust::transform(
+          c2h::device_policy, d_samples.cbegin(), d_samples.cend(), tmp.cbegin(), d_samples.begin(), bit_and_anything{});
+      }
+    }
+  }
+
+  auto h_samples = c2h::host_vector<SampleT>(d_samples);
+
+  // Allocate output histogram
+  auto d_histogram = array<c2h::device_vector<CounterT>, ActiveChannels>();
+  for (size_t c = 0; c < ActiveChannels; ++c)
+  {
+    d_histogram[c].resize(num_levels[c] - 1);
+  }
+
+  SECTION("HistogramEven")
+  {
+    // Setup levels
+    const auto levels       = setup_bin_levels_for_even(num_levels, max_level, max_level_count);
+    const auto& lower_level = levels[0]; // TODO(bgruber): use structured bindings in C++17
+    const auto& upper_level = levels[1];
+    CAPTURE(lower_level, upper_level);
+
+    // Compute reference result
+    auto fp_scales = array<LevelT, ActiveChannels>{}; // only used when LevelT is floating point
+    cs::ignore     = fp_scales; // casting to void was insufficient. TODO(bgruber): use [[maybe_unsued]] in C++17
```

Comment:
@miscco using `cuda::std::ignore =` does produce the following error on CTK 11.1:
```
/home/coder/cccl/cub/test/catch2_test_device_histogram.cu(281): error #174-D: expression has no effect
```
Using `std::ignore = ` produces no such warning.